localstack/services/sns/sns_listener.py

```diff
@@ -249,28 +243,46 @@ def return_response(self, method, path, data, headers, response):
     def should_persist(self, method, path, data, headers, response):
         req_params = parse_request_data(method, path, data)
         action = req_params.get('Action', '')
-        if action in SKIP_PERSISTENCE_ACTIONS:
+        if action in sns_backend.skip_persistence_actions:
             return False
         return super(ProxyListenerSNS, self).should_persist(method, path, data, headers, response)
 
 
+def patch_moto():
+    def patch_create_platform_endpoint(self, *args):
+        try:
+            return create_platform_endpoint_orig(self, *args)
+        except DuplicateSnsEndpointError:
+            custom_user_data, token = args[2], args[3]
+            for endpoint in self.platform_endpoints.values():
+                if endpoint.token == token:
+                    if custom_user_data and custom_user_data != endpoint.custom_user_data:
+                        raise DuplicateSnsEndpointError('Endpoint already exist for token: %s with different attributes'
+                             % token)
+                    else:
```

Comment:
nitpick: `else` technically not required here (as the `if` branch `raise`s and error).